perf(external-share): Port to Entity and use snowflake Ids

CarlSchwan opened the pull request:
- Port away from Files::buildNotExistingFileName
- Use IUser and IGroup instead of plain string


* Resolves: # <!-- related github issue -->

## Summary


## TODO

- [ ] ...

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)
- [x] [Labels added](https://github.com/nextcloud/server/labels) where applicable (ex: bug/enhancement, `3. to review`, feature component)
- [x] [Milestone added](https://github.com/nextcloud/server/milestones) for target branch/version (ex: 32.x for `stable32`)

CarlSchwan committed:
refactor(external-share): Cleanup OCA\FIles_Sharing\External\Manager

- Port away from Files::buildNotExistingFileName
- Use IUser and IGroup instead of plain string

Signed-off-by: Carl Schwan <carl.schwan@nextcloud.com>

CarlSchwan committed:
refactor(OC_Helper): remove buildNotExistingFileNameForView

Move the functionality in the last place it is used OC\Files\Node\Folder

Signed-off-by: Carl Schwan <carl.schwan@nextcloud.com>

CarlSchwan committed:
fix(sharing): Adapt Hooks to new API

Signed-off-by: Carl Schwan <carl.schwan@nextcloud.com>

come-nc commented:
According to integation tests you broke federated shares @CarlSchwan

icewind1991 commented on apps/files_sharing/lib/External/Manager.php:
```diff
@@ -528,7 +538,11 @@ protected function tryOCMEndPoint($remoteDomain, $token, $remoteId, $feedback) {
-				return $this->cloudFederationProviderManager->sendNotification($remoteDomain, $notification);
+				try {
+					$response = $this->cloudFederationProviderManager->sendCloudNotification($remoteDomain, $notification);
+				} catch (OCMProviderException) {
```
maybe add this `catch` block around the switch and have it `return false` directly instead of relying on the `$response` staying `null` bahavior.

CarlSchwan replied:
reverted the whole part, since I suspect this is breaking the integration test

come-nc replied:
still failing :sob:

CarlSchwan committed:
fix: Revert change to the cloud integration

This might fixes the test.

Signed-off-by: Carl Schwan <carl.schwan@nextcloud.com>

CarlSchwan committed:
refactor: use DI-injected logger instead of Server::get

Signed-off-by: Carl Schwan <carl.schwan@nextcloud.com>

CarlSchwan committed:
perf(external-sharing): Port to Entity and SnowflakeId

This removes all the read after write and we don't need to queries all
the time the same share in the same request anymore.

Signed-off-by: Carl Schwan <carl.schwan@nextcloud.com>

CarlSchwan committed:
fix: Remove auto increment from share_external

Signed-off-by: Carl Schwan <carl.schwan@nextcloud.com>

CarlSchwan committed:
refactor(external-share): Port more code to string as type for the id

Signed-off-by: Carl Schwan <carl.schwan@nextcloud.com>

CarlSchwan committed:
refactor(integration): Refactor and fix federation integration tests

Signed-off-by: Carl Schwan <carl.schwan@nextcloud.com>

CarlSchwan committed:
refactor: Add more typing to integration context code

Signed-off-by: Carl Schwan <carl.schwan@nextcloud.com>

CarlSchwan committed:
refactor: Rename some variables in external share's manager

Signed-off-by: Carl Schwan <carl.schwan@nextcloud.com>

CarlSchwan committed:
fix(openapi): Fix returned types from external share

We always been returning an int for for accepted, but since it was
returned as a mixed from the DB, psalm never complained about the fact
this was typed as a bool in the API doc.

Signed-off-by: Carl Schwan <carl.schwan@nextcloud.com>
